src/plugins/es_ui_shared/docs/docs/form_lib/core/form_hook.md

```diff
@@ -0,0 +1,208 @@
+    setIsSubmitted(true);
+    setIsSubmitting(true);
+
+    // If the "isValid" state is "undefined" (=== not all the fields are dirty),
+    // call validate() to run validation on all the fields.
+    const isValid = form.isValid ?? (await form.validate());
+    setIsSubmitting(false);
+
+    if (isValid) {
+      console.log('Form data:', form.getFormData());
+    }
+  };
+
+  const hasErrors = isSubmitted && form.isValid === false;
+
+  return (
+    <Form form={form}>
+      <UseField path="firstName" config={{ validations: [{ validator }] }} />
+      <UseField path="lastName" />
+
+      <button
+        onClick={onClickSubmit}
+        disabled={isSubmitting || hasErrors}
+      >
+        {isSubmitting ? 'Sending...' : 'Submit'}
+      </button>
+      {hasErrors && <div>Form is invalid.</div>}
+    </Form>
+  );
+};
+```
+
+### id
+
+**Type:** `string`
+
+The form id. If none was provided, "default" will be returned.
+
+## Handlers
+
+### submit()
+
+**Returns:** `Promise<{ data: T | {}, isValid: boolean }>`
+
+This handlers submits the form and returns its data and validity. If the form is not valid, the data will be `null` as only valid data is passed through the `serializer(s)` before being returned.
+
+```js
+const { data, isValid } = await form.submit();
+```
+
+### validate()
+
+**Returns:** `Promise<boolean>`
+
+Use this handler to get the validity of the form.
+
+```js
+const isFormValid = await form.validate();
+```
+
+### getFields()
+
+**Returns:** `{ [path: string]: FieldHook }`
+
+Access any field on the form.
+
+```js
+const { name: nameField }  = form.getFields();
+```
+
+### getFormData()
+
+**Arguments:** `options?: { unflatten?: boolean }`  
+**Returns:** `T | R`
+
+Return the form data. It accepts an optional options with an `unflatten` parameter (defaults to `true`). If you are only interested in the raw form data, pass `unflatten: false` to the handler ([read more about the "Raw" data state here](in_out_raw_state.md#raw-data-state)).
+
+```js
+const formData  = form.getFormData();
+
+const rawFormData  = form.getFormData({ unflatten: false });
+```
+
+### getErrors()
+
+**Returns:** `string[]`
+
+Returns an array with of all errors in the form.
+
+```js
+const errors  = form.getErrors();
+```
+
+### reset()
+
+**Arguments:** `options?: { resetValues?: boolean; defaultValue?: any }` 
+
+Resets the form to its initial state. It accetps an optional object of configuration:
+
+- `resetValues` (default: `true`). Flag to indicate if we want to not only reset the form state (`isValid`, `isSubmitted`...) but also the field values. If set to `true` will put back the default value passed to the form or declared on the field config (in that order).
+
+- `defaultValue`. In some cases you might not want to reset the form to the default value initiallly provided to the form (probably because it is data that came from the server and you want a clean form). In this case you can provide a new `defaultValue` object when resetting.
```

Comment:
Might also be worth adding a word of caution here that using this to imperatively set form state can lead to users form data being lost and that this is a kind of escape hatch.

Reply:
Maybe we can even call it something like `dangerouslySetDefaultValues` if we feel strongly enough 😄

Reply:
I am not sure I follow :blush: If the consumer specifically calls `form.reset()` in his code, isn't it implicit that the values will be all changed (and lost)?